sunpy/net/dataretriever/sources/swap.py

```diff
@@ -0,0 +1,107 @@
+''' SWAPClient definition '''
+
+__author__ = "Sudarshan Konge"
+__email__ = "sudk1896@gmail.com"
+
+import datetime
+import urllib2
+
+
+from astropy import units as u
+from bs4 import BeautifulSoup
+from sunpy.net.dataretriever.client import GenericClient
+from sunpy.util.scraper import Scraper
+
+from sunpy.time import TimeRange
+
+__all__ = ['SWAPClient']
+
+class SWAPClient(GenericClient):
+    
+    """ Returns a list of URLS to Proba2 SWAP files corresponding to value of input timerange.
+    URL source: http://proba2.oma.be/swap/data/bsd/
+
+    The earliest date available is from 24-Nov-2009
+
+    Parameters
+    ----------
+    timerange: sunpy.time.TimeRange
+        time range for which data is to be downloaded.
+        Example value - TimeRange('2015-12-30 00:00:00','2015-12-31 00:01:00')
+    
+    Instrument: Fixed argument = 'swap'
+
+    Returns
+    -------
+    urls: list
+        list of urls corresponding to requested time range.
+
+    Examples
+    --------
+    >>> from sunpy.time.timerange import TimeRange
+    >>> from sunpy.net.vso.attrs import Time,Instrument,Source,Level
+    >>> from sunpy.net.dataretriever.client import QueryResponse
+    >>> from sunpy.net.dataretriever.sources import swap
+
+    >>> LCClient = swap.SWAPClient()
+
+    >>> qr = LCClient.query(Time('2015-12-30 00:00:00','2015-12-31 00:05:00'),Instrument('swap'))
+    >>> res = LCClient.get(qr)
+    >>> dl = res.wait()
+    """
+    
+    def _get_url_for_timerange(cls, timerange, **kwargs):
+        """ returns list of urls corresponding
+        to given TimeRange. """
+
+        if timerange.start < datetime.datetime(2009,11,24):
+            raise ValueError('Earliest date for which SWAP data is available is 2009-11-24')
```

Comment:
define the datetime outside and then use it twice, so we don't need to change it if it becomes available (probably won't happen but it would be easier to do if so).

```python
SWAP_STARTDATE = datetime.datetime(2009, 11, 24)
...
if timerange.start < SWAP_STARTDATE:
    raise ValueError('Earliest date for which SWAP data is available is {0:%Y-%m-%d}'.format(SWAP_STARTDATE))
```